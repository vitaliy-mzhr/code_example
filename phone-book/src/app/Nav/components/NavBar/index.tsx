import * as React from 'react';
import NavBarStyles from './styles';
import { AppBar, Toolbar, Input, Typography, Tooltip, IconButton } from '@material-ui/core';
import { Search } from '@material-ui/icons';
import { NavButton } from '../../components';
import { Link } from 'react-router-dom';

interface AppNavBarProps {
  isMainPage: boolean;
  isOnSearch: boolean;
  handleRedirect: () => void;
  handleSearch: (e: any) => void;
  inputValue: string;
  clearInput: () => void;
}

const NavBar: React.StatelessComponent<AppNavBarProps> = (
  {isOnSearch, handleRedirect, inputValue, handleSearch, clearInput, isMainPage}) => (
    <NavBarStyles>
      <AppBar position="static">
        <Toolbar className="app-bar">
          <NavButton clearInput={clearInput} showBackArrow={!isMainPage}/>
          <Typography className="title" variant="title" color="inherit" noWrap={true}>
            Phone-book app
          </Typography>
          <div className="flex-grow"/>
          {isOnSearch ? (
            <div className="search-container">
              <div className="search-icon">
                <Search/>
              </div>
              <Input
                value={inputValue}
                autoFocus={isOnSearch}
                placeholder="Searchâ€¦"
                disableUnderline={true}
                onChange={handleSearch}
                className={`search-input ${isOnSearch ? 'add-transition' : ''}`}
                onClick={handleRedirect}
              />
            </div>
          ) : (
            <Tooltip title={'Search contact'}>
              <Link to={'/search'}>
                <IconButton className="search-button">
                  <Search/>
                </IconButton>
              </Link>
            </Tooltip>
          )}
        </Toolbar>

      </AppBar>
    </NavBarStyles>
  );

export default NavBar;
