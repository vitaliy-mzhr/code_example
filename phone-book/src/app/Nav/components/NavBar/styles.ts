import styled from 'styled-components';

const NavBarStyles = styled.div`
   width: 100%;
   .app-bar {
    min-height: ${props => props.theme.dimensions.navMinHeightRem}rem;
   }
   .back-button {
      color: ${props => props.theme.colors.whiteColor};
   }
   .home-button {
    margin-left: -12px;
    margin-right: 20px;
    color: ${props => props.theme.colors.whiteColor};
   }
   .search-button {
    color: ${props => props.theme.colors.whiteColor};
   }
   .flex-grow {
    flex-grow: 1;
   }
   .search-container {
    position: relative;
    border-radius: ${props => props.theme.borders.mainBorderRadiusPx}px;
    margin-left: 0;
    background-color: ${props => props.theme.colors.navSearchBackground};
    //width: 100%;
    &:hover {
      background-color: ${props => props.theme.colors.navSearchBackgroundHover};
    }
   }
   .search-icon {
    height: 100%;
    position: absolute;
    display: flex;
    align-items: center;
    justify-content: center;
    pointer-events: none;
    width: 4rem;
   }
   .search-input {
    color: inherit;
    padding: .1rem .1rem .1rem 4.1rem;
    input {
      width: 10rem;
    }
    &.add-transition {
      input {
        width: 10rem;
        &:focus {
        width: 15rem;
      }
        transition: width .15s ease-in;
      }
    }
   }
   
   @media all and (max-width: ${props => props.theme.media.mobileMaxWidthPx}px) {
   .app-bar {
    padding: 0 .7rem;
    min-height: ${props => props.theme.dimensions.navMinHeightRem * 0.9}rem;
   }
    .title {
      display: none;
    }
    .search-input {
      input {
        width: 7rem;
     }
     &.add-transition {
      input {
        width: 7rem;
        &:focus {
        width: 10rem;
      }
     }
     }
    }
  }
`;

export default NavBarStyles;
