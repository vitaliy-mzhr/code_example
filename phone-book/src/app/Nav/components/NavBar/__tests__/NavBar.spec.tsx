import * as React from 'react';
import { shallow } from 'enzyme';
import NavBar from '../../NavBar';

interface AppNavBarProps {
  isMainPage: boolean;
  isOnSearch: boolean;
  handleRedirect: () => void;
  handleSearch: (e: any) => void;
  inputValue: string;
  clearInput: () => void;
}

describe('NavBar component test', () => {

  it('NavBar component renders correctly', () => {
    const mockProps: AppNavBarProps = {
      isMainPage: true,
      isOnSearch: false,
      handleRedirect: jest.fn(),
      handleSearch: jest.fn(),
      inputValue: 'sdas',
      clearInput: jest.fn(),
    };
    expect(shallow(<NavBar {...mockProps}/>)).toMatchSnapshot();
  });

  it('NavBar component renders correctly with different props', () => {
    const mockProps: AppNavBarProps = {
      isMainPage: false,
      isOnSearch: true,
      handleRedirect: jest.fn(),
      handleSearch: jest.fn(),
      inputValue: '1',
      clearInput: jest.fn(),
    };
    expect(shallow(<NavBar {...mockProps}/>)).toMatchSnapshot();
  });
});