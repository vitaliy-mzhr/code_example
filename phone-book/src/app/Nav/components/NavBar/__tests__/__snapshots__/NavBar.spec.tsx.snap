// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component test NavBar component renders correctly 1`] = `
<styled.div>
  <WithStyles(AppBar)
    position="static"
  >
    <WithStyles(Toolbar)
      className="app-bar"
    >
      <NavButton
        clearInput={[MockFunction]}
        showBackArrow={false}
      />
      <WithStyles(Typography)
        className="title"
        color="inherit"
        noWrap={true}
        variant="title"
      >
        Phone-book app
      </WithStyles(Typography)>
      <div
        className="flex-grow"
      />
      <WithStyles(Tooltip)
        title="Search contact"
      >
        <Link
          replace={false}
          to="/search"
        >
          <WithStyles(IconButton)
            className="search-button"
          >
            <pure(SearchIcon) />
          </WithStyles(IconButton)>
        </Link>
      </WithStyles(Tooltip)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</styled.div>
`;

exports[`NavBar component test NavBar component renders correctly with different props 1`] = `
<styled.div>
  <WithStyles(AppBar)
    position="static"
  >
    <WithStyles(Toolbar)
      className="app-bar"
    >
      <NavButton
        clearInput={[MockFunction]}
        showBackArrow={true}
      />
      <WithStyles(Typography)
        className="title"
        color="inherit"
        noWrap={true}
        variant="title"
      >
        Phone-book app
      </WithStyles(Typography)>
      <div
        className="flex-grow"
      />
      <div
        className="search-container"
      >
        <div
          className="search-icon"
        >
          <pure(SearchIcon) />
        </div>
        <WithStyles(Input)
          autoFocus={true}
          className="search-input add-transition"
          disableUnderline={true}
          onChange={[MockFunction]}
          onClick={[MockFunction]}
          placeholder="Searchâ€¦"
          value="1"
        />
      </div>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</styled.div>
`;
